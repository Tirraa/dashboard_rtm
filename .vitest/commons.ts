export const INVALID_PATH = './bless/the/draco/these/rounds/holy/sorry/bout/your/luck';

export const FAKE_BLOG_METADATAS_A = {
  'fake-category-one': {
    'fake-subcategory-one': {
      DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'],
      en: ['fake-post-01', 'fake-post-03'],
      it: ['fake-post-01', 'fake-post-02'],
      fr: ['fake-post-01']
    },
    'fake-subcategory-three': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], es: ['fake-post-01'], zh: ['fake-post-02'] },
    'fake-subcategory-two': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], fr: ['fake-post-02'] }
  },
  'fake-category-three': {
    'fake-subcategory-five': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], en: ['fake-post-03'], it: ['fake-post-03'] },
    'fake-subcategory-four': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], es: ['fake-post-04'] }
  },
  'fake-category-two': { 'fake-subcategory-six': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], zh: ['fake-post-10'] } }
};

export const FAKE_BLOG_METADATAS_B = {
  'fake-category-one': {
    'fake-subcategory-one': {
      DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'],
      en: ['fake-post-01', 'fake-post-03'],
      it: ['fake-post-01', 'fake-post-02'],
      fr: ['fake-post-01']
    },
    'fake-subcategory-three': {
      DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'],
      es: ['fake-post-01'],
      zh: ['fake-post-02']
    },
    'fake-subcategory-two': { fr: ['fake-post-02'] }
  },
  'fake-category-three': {
    'fake-subcategory-five': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], en: ['fake-post-03'], it: ['fake-post-03'] },
    'fake-subcategory-four': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], es: ['fake-post-04'] }
  },
  'fake-empty-subcategories-nested-in-category': {
    'fake-subcategory-seven': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], zh: ['fake-post-10'] },
    'fake-empty-subcategory': {}
  },
  'fake-category-two': { 'fake-subcategory-six': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], zh: ['fake-post-10'] } },
  'fake-empty-category': {}
};
