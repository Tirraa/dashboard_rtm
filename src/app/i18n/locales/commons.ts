export namespace CommonsVocab {
  export const brand = 'Rust Team Management';
  export const logo = 'logo'; // {ToDo} i18n this!
}

export default CommonsVocab;
