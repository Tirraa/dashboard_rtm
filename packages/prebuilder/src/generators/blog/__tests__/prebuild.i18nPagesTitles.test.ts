// eslint-disable-next-line import/no-extraneous-dependencies
import { FAKE_BLOG_METADATAS_A, FAKE_BLOG_METADATAS_B } from '𝕍/commons';
// eslint-disable-next-line import/no-extraneous-dependencies
import { afterAll, describe, expect, it } from 'vitest';

import generateI18nPagesTitles from '../i18nPagesTitles';

const fs = require('fs/promises');

const __TARGET_FOLDER_ROOT = './packages/prebuilder/src/generators/blog/__tests__/FAKE_CODEGEN';
const __TARGET_FOLDER = __TARGET_FOLDER_ROOT + '/' + 'I18N_PAGES_TITLES';

describe('generateI18nPagesTitles (formatted)', () => {
  afterAll(async () => {
    await fs.rm(__TARGET_FOLDER, { recursive: true });
  });

  const pretty = true;

  it('should match snapshot', async () => {
    const targetFile = 'EMPTY_FAKE_PAGES_TITLES';

    await generateI18nPagesTitles({}, pretty, targetFile, __TARGET_FOLDER);
    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });

  it('should match snapshot', async () => {
    const targetFile = 'FAKE_PAGES_TITLES_FAKE_METADATAS_A';

    await generateI18nPagesTitles(FAKE_BLOG_METADATAS_A, pretty, targetFile, __TARGET_FOLDER);
    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });

  it('should match snapshot', async () => {
    const targetFile = 'FAKE_PAGES_TITLES_FAKE_METADATAS_B';

    await generateI18nPagesTitles(FAKE_BLOG_METADATAS_B, pretty, targetFile, __TARGET_FOLDER);
    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });
});

describe('generateI18nPagesTitles (ugly)', () => {
  afterAll(async () => {
    await fs.rm(__TARGET_FOLDER, { recursive: true });
  });

  const pretty = false;

  it('should match snapshot', async () => {
    const targetFile = 'EMPTY_FAKE_PAGES_TITLES';

    await generateI18nPagesTitles({}, pretty, targetFile, __TARGET_FOLDER);
    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });

  it('should match snapshot', async () => {
    const targetFile = 'FAKE_PAGES_TITLES_FAKE_METADATAS_A';

    await generateI18nPagesTitles(FAKE_BLOG_METADATAS_A, pretty, targetFile, __TARGET_FOLDER);
    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });

  it('should match snapshot', async () => {
    const targetFile = 'FAKE_PAGES_TITLES_FAKE_METADATAS_B';

    await generateI18nPagesTitles(FAKE_BLOG_METADATAS_B, pretty, targetFile, __TARGET_FOLDER);
    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });
});
