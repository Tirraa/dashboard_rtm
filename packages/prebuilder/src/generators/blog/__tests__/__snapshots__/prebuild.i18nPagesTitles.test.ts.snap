// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateI18nPagesTitles (formatted) > should match snapshot 1`] = `
"// AUTOGENERATED: Don't edit this file!
const _: EmptyString = '';
const EMPTY_FAKE_PAGES_TITLES = {} as const;
type EmptyString = '';
export default EMPTY_FAKE_PAGES_TITLES;
"
`;

exports[`generateI18nPagesTitles (formatted) > should match snapshot 2`] = `
"// AUTOGENERATED: Don't edit this file!
const _: EmptyString = '';
const FAKE_PAGES_TITLES_FAKE_METADATAS_A = {
  "fake-category-one": _,
  "fake-subcategory-one": _,
  "fake-subcategory-three": _,
  "fake-subcategory-two": _,
  "fake-category-three": _,
  "fake-subcategory-five": _,
  "fake-subcategory-four": _,
  "fake-category-two": _,
  "fake-subcategory-six": _
} as const;
type EmptyString = '';
export default FAKE_PAGES_TITLES_FAKE_METADATAS_A;
"
`;

exports[`generateI18nPagesTitles (formatted) > should match snapshot 3`] = `
"// AUTOGENERATED: Don't edit this file!
const _: EmptyString = '';
const FAKE_PAGES_TITLES_FAKE_METADATAS_B = {
  "fake-category-one": _,
  "fake-subcategory-one": _,
  "fake-subcategory-three": _,
  "fake-subcategory-two": _,
  "fake-category-three": _,
  "fake-subcategory-five": _,
  "fake-subcategory-four": _,
  "fake-empty-subcategories-nested-in-category": _,
  "fake-subcategory-seven": _,
  "fake-empty-subcategory": _,
  "fake-category-two": _,
  "fake-subcategory-six": _,
  "fake-empty-category": _
} as const;
type EmptyString = '';
export default FAKE_PAGES_TITLES_FAKE_METADATAS_B;
"
`;

exports[`generateI18nPagesTitles (ugly) > should match snapshot 1`] = `
"// AUTOGENERATED: Don't edit this file!
const _: EmptyString = '';
const EMPTY_FAKE_PAGES_TITLES = {} as const;
type EmptyString = '';
export default EMPTY_FAKE_PAGES_TITLES;"
`;

exports[`generateI18nPagesTitles (ugly) > should match snapshot 2`] = `
"// AUTOGENERATED: Don't edit this file!
const _: EmptyString = '';
const FAKE_PAGES_TITLES_FAKE_METADATAS_A = {"fake-category-one":_,"fake-subcategory-one":_,"fake-subcategory-three":_,"fake-subcategory-two":_,"fake-category-three":_,"fake-subcategory-five":_,"fake-subcategory-four":_,"fake-category-two":_,"fake-subcategory-six":_} as const;
type EmptyString = '';
export default FAKE_PAGES_TITLES_FAKE_METADATAS_A;"
`;

exports[`generateI18nPagesTitles (ugly) > should match snapshot 3`] = `
"// AUTOGENERATED: Don't edit this file!
const _: EmptyString = '';
const FAKE_PAGES_TITLES_FAKE_METADATAS_B = {"fake-category-one":_,"fake-subcategory-one":_,"fake-subcategory-three":_,"fake-subcategory-two":_,"fake-category-three":_,"fake-subcategory-five":_,"fake-subcategory-four":_,"fake-empty-subcategories-nested-in-category":_,"fake-subcategory-seven":_,"fake-empty-subcategory":_,"fake-category-two":_,"fake-subcategory-six":_,"fake-empty-category":_} as const;
type EmptyString = '';
export default FAKE_PAGES_TITLES_FAKE_METADATAS_B;"
`;
